### DISCUSSION TITLE
Local database update and/or rebuild entire database not working properly in 20230630

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1386] [Fixed Issues]
Running rAudio 20230630

I added three albums to my library and proceeded to update / changed files only.

The end result was:
1) An extremely high number in the Library / Lastest count (460 to be exact).
2) Viewing the contents of "Latest" showed the 460 albums mostly with the default cover art (Note: the coverart for these albums where displayed correctly in rAudio 20230620).   I also confirmed that the coverart files where still present in my physical NAS library.
3) Viewing an album in "Latest" did not show any tracks (only the default thumbnail).

I then rebuilt the entire database with the same end result as described above.

This is most likely the same issue as #1379 as the details are extremely similar.

I reproduce the same issue on a Raspberry Pi 3B+ Rev 1.3 and Raspberry Pi 3B Rev 1.2

Fortunately I saved a backup prior to the update and could easily restore a backup / library data only which worked perfect (nice feature by the way).

I also built an system directly from i20230620 and rebuilt the database **with NO issues**.

This bug was definitely introduce in 20230630.

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

Jim

**Comment by ccaper** [comment]
Thank you for posting the details, and linking to my bug report.  I have now reverted to 20230620, rebuilt my entire database and updated thumbnails with no issue.  Took the 20230630 update, rebuilt database and updated thumbnails, and have the problem again.  Its definitely a problem with the 20230630 release, and have now gone back to the 20230620 release.

I've now ordered a 2nd SD card, cloned my original, and using the 2nd SD card to try updates before going forward to be safe.

I looked over the code changes between the 20230620 and the 20230630 releases, but nothing has jumped out at me as the possible culprit.

**Comment by klassikmann** [comment]
Perhaps, https://github.com/rern/rAudio/discussions/1377 is related to your problem.
A fraction of time is gone for a local database update using `mpc rescan --wait <path>` compared to "Update database" in the library view.

**Reply by Sagnasty** [reply]
Possibly.   I was most concerned with trying to determine in what release the problem was introduce (and if I could find a workaround in the interim).

**Comment by Sagnasty** [comment]
I rebuilt all my systems directly from image 20231022 with no further issues with library updates.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+
